src/Facets/SymbiosisFacet.sol

```diff
@@ -0,0 +1,132 @@
+import { Validatable } from "../Helpers/Validatable.sol";
+
+/// @title Symbiosis Facet
+/// @author Symbiosis (https://symbiosis.finance)
+/// @notice Provides functionality for bridging through Symbiosis Protocol
+/// @custom:version 1.0.0
+contract SymbiosisFacet is ILiFi, ReentrancyGuard, SwapperV2, Validatable {
+    /// @notice The contract address of the Symbiosis router on the source chain
+    ISymbiosisMetaRouter private immutable symbiosisMetaRouter;
+    address private immutable symbiosisGateway;
+
+    /// Types ///
+    struct SymbiosisData {
+        bytes firstSwapCalldata;
+        bytes secondSwapCalldata;
+        address intermediateToken;
+        address firstDexRouter;
+        address secondDexRouter;
+        address[] approvedTokens;
+        address callTo; // bridging entrypoint
+        bytes callData; // bridging calldata
+    }
+
+    /// Constructor ///
+
+    /// @notice Initialize the contract.
+    /// @param _symbiosisMetaRouter The contract address of the Symbiosis MetaRouter on the source chain.
+    /// @param _symbiosisGateway The contract address of the Symbiosis Gateway on the source chain.
+    constructor(
+        ISymbiosisMetaRouter _symbiosisMetaRouter,
+        address _symbiosisGateway
+    ) {
+        symbiosisMetaRouter = _symbiosisMetaRouter;
+        symbiosisGateway = _symbiosisGateway;
+    }
+
+    /// @dev Contains the business logic for the bridge via Symbiosis
+    /// @param _bridgeData the core information needed for bridging
+    /// @param _symbiosisData data specific to Symbiosis
+    function _startBridge(
+        ILiFi.BridgeData memory _bridgeData,
+        SymbiosisData calldata _symbiosisData
+    ) internal {
+        bool isNative = LibAsset.isNativeAsset(_bridgeData.sendingAssetId);
+        uint256 nativeAssetAmount;
+
+        if (isNative) {
+            nativeAssetAmount = _bridgeData.minAmount;
+        } else {
+            LibAsset.maxApproveERC20(
+                IERC20(_bridgeData.sendingAssetId),
+                symbiosisGateway,
+                _bridgeData.minAmount
+            );
+        }
+
+        symbiosisMetaRouter.metaRoute{ value: nativeAssetAmount }(
+            ISymbiosisMetaRouter.MetaRouteTransaction(
+                _symbiosisData.firstSwapCalldata,
+                _symbiosisData.secondSwapCalldata,
+                _symbiosisData.approvedTokens,
+                _symbiosisData.firstDexRouter,
+                _symbiosisData.secondDexRouter,
+                _bridgeData.minAmount,
+                isNative,
+                _symbiosisData.callTo,
+                _symbiosisData.callData
+            )
+        );
+
+        emit LiFiTransferStarted(_bridgeData);
+    }
+
+    /// External Methods ///
+
+    /// @notice Bridges tokens via Symbiosis
+    /// @param _bridgeData The core information needed for bridging
+    /// @param _symbiosisData The data specific to Symbiosis
+    function startBridgeTokensViaSymbiosis(
```

Comment:
The order of functions in the contract is a bit funny (internal helper methods before external)